src/spikeinterface/widgets/unit_waveforms.py

```diff
@@ -225,9 +241,71 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
                     xvec = xvectors_flat + i * 0.7 * dp.delta_x
                 else:
                     xvec = xvectors_flat
-
+                # plot template shading if waveforms are not plotted
+                if not dp.plot_waveforms and dp.do_shading:
+                    if len(dp.templates_shading) == 2:
+                        lower_bound = (
+                            dp.templates_shading[0][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
+                        upper_bound = (
+                            dp.templates_shading[1][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
+                        ax.fill_between(
+                            xvec,
+                            lower_bound.T.flatten(),
+                            upper_bound.T.flatten(),
+                            color="gray",
+                            alpha=0.5,
+                        )
+                    elif len(dp.templates_shading) == 4:
+                        lower_bound = (
+                            dp.templates_shading[0][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
+                        mid_lower_bound = (
+                            dp.templates_shading[1][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
+                        mid_upper_bound = (
+                            dp.templates_shading[2][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
+                        upper_bound = (
+                            dp.templates_shading[3][i, :, :][:, chan_inds] * dp.y_scale + dp.y_offset[:, chan_inds]
+                        )
```

Comment:
This could be a general case of N must be even. and written in a loop

Reply:
done